app/scripts/store/dimStores.directive.js

```diff
@@ -89,6 +92,13 @@
       vm.vault = dimStoreService.getVault();
     });
 
+    // Update the search service when a filter has been applied. We will need this to:
+    //   A) see if there is an active filter (non-empty query string)
+    //   B) populate the item count for each category
+    $scope.$on('dim-filtered', function(e) {
+      vm.search = dimSearchService;
+    });
```

Comment:
I don't think you need this event - `dimSearchService` will be watched automatically.